packer-variables/default-args.j2

```diff
@@ -40,13 +40,18 @@
     "custom_role": "true",
     {# If a custom role is create in a different location append that
     location after mounting that Path using docker #}
-    "custom_role_names": "/image-builder/images/capi/image/ansible",
+    {% set custom_role_names = "/image-builder/images/capi/image/ansible" %}
     {# Update the required packer based on the OS consumption #}
     {% if os_type == "photon-3" %}
     "distro_version": "3.0",
-    "extra_rpms": "glibc zlib filesystem ethtool pkg-config bash bzip2 shadow procps-ng iana-etc coreutils bc libtool findutils xz iproute2 util-linux kmod linux-esx iptables Linux-PAM systemd dbus file e2fsprogs rpm gawk cloud-utils gptfdisk nfs-utils openssh gdbm photon-release photon-repos haveged sed grep cpio gzip vim libdb tdnf less iputils bindutils diffutils bridge-utils cri-tools apparmor-utils apparmor-profiles krb5 which tzdata motd sudo iotop lsof traceroute ethtool dstat ltrace ipset netcat tcpdump wget"
+    "extra_rpms": "glibc zlib filesystem ethtool pkg-config bash bzip2 shadow procps-ng iana-etc coreutils bc libtool findutils xz iproute2 util-linux kmod linux-esx iptables Linux-PAM systemd dbus file e2fsprogs rpm gawk cloud-utils gptfdisk nfs-utils openssh gdbm photon-release photon-repos haveged sed grep cpio gzip vim libdb tdnf less iputils bindutils diffutils bridge-utils cri-tools apparmor-utils apparmor-profiles krb5 which tzdata motd sudo iotop lsof traceroute ethtool dstat ltrace ipset netcat tcpdump wget",
     {% elif os_type == "ubuntu-2004-efi" %}
     "extra_debs": "unzip iptables-persistent nfs-common",
-    "boot_disable_ipv6": "1"
+    "boot_disable_ipv6": "1",
     {% endif %}
+    {% if photon3_stig_compliance == "true" %}
```

Comment:
As we are unable to use set because `ansible_user_vars` is a templated string (One solution is rendering the output again using Jinja which seems more complicated to me 😮‍💨 )

Can we add the conditional check in the string itself something like the one below (at least this avoids updating the values at two different places 🤔 )
```
{% set photon3_stig_compliance_data ="run_sysctl_net_ipv6_conf_all_mc_forwarding=false run_sysctl_net_ipv6_conf_default_mc_forwarding=false run_sysctl_net_ipv6_conf_eth0_mc_forwarding=false run_sshd_banner=false" %}
"ansible_user_vars": "{{ random_templated_data }} {% if photon3_stig_compliance %}{{ photon3_stig_compliance_data }}{% endif %} photon3_stig_compliance={{ photon3_stig_compliance }}",
```

Upto you if you want to convert the `custom_role_names` also to a similar approach as above.

Later we can remove these conditional checks once.

Reply:
Able to get the desired values for both 1.25.7 and 1.24.9